# 1. Data import & raw data information.

In [1]:
import pandas as pd
import numpy as np
pd.set_option("display.max_columns", None)

In [2]:
df=pd.read_csv(r'C:\Users\alexd\Ironhack\Proyectos\Project2_DA_Ironhack\Raw - Data\w3-Database-Project-main\data\film.csv')

df.head()

film_id             title  \
0        1  ACADEMY DINOSAUR   
1        2    ACE GOLDFINGER   
2        3  ADAPTATION HOLES   
3        4  AFFAIR PREJUDICE   
4        5       AFRICAN EGG   

                                         description  release_year  \
0  A Epic Drama of a Feminist And a Mad Scientist...          2006   
1  A Astounding Epistle of a Database Administrat...          2006   
2  A Astounding Reflection of a Lumberjack And a ...          2006   
3  A Fanciful Documentary of a Frisbee And a Lumb...          2006   
4  A Fast-Paced Documentary of a Pastry Chef And ...          2006   

   language_id  original_language_id  rental_duration  rental_rate  length  \
0            1                   NaN                6         0.99      86   
1            1                   NaN                3         4.99      48   
2            1                   NaN                7         2.99      50   
3            1                   NaN                5         2.99     117   
4            1                   NaN                6         2.99     130   

   replacement_cost rating                  special_features  \
0             20.99     PG  Deleted Scenes,Behind the Scenes   
1             12.99      G           Trailers,Deleted Scenes   
2             18.99  NC-17           Trailers,Deleted Scenes   
3             26.99      G    Commentaries,Behind the Scenes   
4             22.99      G                    Deleted Scenes   

           last_update  
0  2006-02-15 05:03:42  
1  2006-02-15 05:03:42  
2  2006-02-15 05:03:42  
3  2006-02-15 05:03:42  
4  2006-02-15 05:03:42

In [3]:
dim_orig=df.shape
dim_orig

(1000, 13)

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 13 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   film_id               1000 non-null   int64  
 1   title                 1000 non-null   object 
 2   description           1000 non-null   object 
 3   release_year          1000 non-null   int64  
 4   language_id           1000 non-null   int64  
 5   original_language_id  0 non-null      float64
 6   rental_duration       1000 non-null   int64  
 7   rental_rate           1000 non-null   float64
 8   length                1000 non-null   int64  
 9   replacement_cost      1000 non-null   float64
 10  rating                1000 non-null   object 
 11  special_features      1000 non-null   object 
 12  last_update           1000 non-null   object 
dtypes: float64(3), int64(5), object(5)
memory usage: 101.7+ KB


# 2. Data Cleaning.
### Null Values

In [5]:
nan_cols=df.isnull().sum()
nan_cols[nan_cols>0]

original_language_id    1000
dtype: int64

In [6]:
del df['original_language_id']

In [7]:
df.shape

(1000, 12)

# 3.Determination of data type per column:

**title** --> Ok, it is an object(string) type. No clenaning needs to be done in this column.

**description** --> Ok, it is an object(string) type. No clenaning needs to be done in this column.

**release_year** --> All of them are 2006. OK the column to be an integer.

**language_id** --> All of them are language id 1. OK the column to be an integer.

**rental_duration** --> This takes unique values from 3 to 7 days. OK the column to be an integer.

**rental_rate** --> This takes unique values [0.99, 4.99, 2.99]. OK the column to be floats.

**length** --> Takes unique values as minutes. ok the column to be an integer.

**replacement_cost** --> This takes unique values with .99. OK the column to be floats.

**rating** --> Ok, it is an object(string) type. No clenaning needs to be done in this column.

**special_features** --> Ok, it is an object(string) type. No clenaning needs to be done in this column.

**last_update** --> This must be changed into datetime.


In [8]:
df.title.unique()

array(['ACADEMY DINOSAUR', 'ACE GOLDFINGER', 'ADAPTATION HOLES',
       'AFFAIR PREJUDICE', 'AFRICAN EGG', 'AGENT TRUMAN',
       'AIRPLANE SIERRA', 'AIRPORT POLLOCK', 'ALABAMA DEVIL',
       'ALADDIN CALENDAR', 'ALAMO VIDEOTAPE', 'ALASKA PHANTOM',
       'ALI FOREVER', 'ALICE FANTASIA', 'ALIEN CENTER', 'ALLEY EVOLUTION',
       'ALONE TRIP', 'ALTER VICTORY', 'AMADEUS HOLY',
       'AMELIE HELLFIGHTERS', 'AMERICAN CIRCUS', 'AMISTAD MIDSUMMER',
       'ANACONDA CONFESSIONS', 'ANALYZE HOOSIERS', 'ANGELS LIFE',
       'ANNIE IDENTITY', 'ANONYMOUS HUMAN', 'ANTHEM LUKE',
       'ANTITRUST TOMATOES', 'ANYTHING SAVANNAH', 'APACHE DIVINE',
       'APOCALYPSE FLAMINGOS', 'APOLLO TEEN', 'ARABIA DOGMA',
       'ARACHNOPHOBIA ROLLERCOASTER', 'ARGONAUTS TOWN', 'ARIZONA BANG',
       'ARK RIDGEMONT', 'ARMAGEDDON LOST', 'ARMY FLINTSTONES',
       'ARSENIC INDEPENDENCE', 'ARTIST COLDBLOODED', 'ATLANTIS CAUSE',
       'ATTACKS HATE', 'ATTRACTION NEWTON', 'AUTUMN CROW', 'BABY HALL',
       'BACKLASH UND

In [9]:
df.description.unique()

array(['A Epic Drama of a Feminist And a Mad Scientist who must Battle a Teacher in The Canadian Rockies',
       'A Astounding Epistle of a Database Administrator And a Explorer who must Find a Car in Ancient China',
       'A Astounding Reflection of a Lumberjack And a Car who must Sink a Lumberjack in A Baloon Factory',
       'A Fanciful Documentary of a Frisbee And a Lumberjack who must Chase a Monkey in A Shark Tank',
       'A Fast-Paced Documentary of a Pastry Chef And a Dentist who must Pursue a Forensic Psychologist in The Gulf of Mexico',
       'A Intrepid Panorama of a Robot And a Boy who must Escape a Sumo Wrestler in Ancient China',
       'A Touching Saga of a Hunter And a Butler who must Discover a Butler in A Jet Boat',
       'A Epic Tale of a Moose And a Girl who must Confront a Monkey in Ancient India',
       'A Thoughtful Panorama of a Database Administrator And a Mad Scientist who must Outgun a Mad Scientist in A Jet Boat',
       'A Action-Packed Tale of a Man 

In [10]:
df.release_year.unique()   # all of them from 2006

array([2006], dtype=int64)

In [11]:
df.language_id.unique()   # all of them are same language

array([1], dtype=int64)

In [12]:
df.rental_duration.unique()   # in days

array([6, 3, 7, 5, 4], dtype=int64)

In [13]:
df.rental_rate.unique()  # rate from 0 to 5.

array([0.99, 4.99, 2.99])

In [14]:
df.length.unique()  # this are minutes.

array([ 86,  48,  50, 117, 130, 169,  62,  54, 114,  63, 126, 136, 150,
        94,  46, 180,  82,  57, 113,  79, 129,  85,  92, 181,  74, 179,
        91, 168, 119, 153, 147, 127, 121,  68,  99, 148, 137, 170,  83,
       108, 118, 162, 182,  75, 173,  87,  65,  90, 122, 160,  89, 175,
       106,  73, 151, 100,  53,  77, 142,  93, 163, 103,  61,  71, 102,
        76,  98,  72, 176, 123,  56, 161,  60, 133, 125,  67,  52, 120,
       167,  70, 135, 112, 183, 110, 152,  51, 146,  66, 185, 124, 107,
       101, 132, 143, 165,  81, 164,  58,  95, 139,  55, 149, 109,  59,
       172, 115, 184, 166,  64, 104,  69,  84, 144,  88, 141,  47,  49,
       177, 154, 159, 178,  96, 116, 171, 140, 155, 158, 174, 138,  97,
       131, 156,  80, 145, 111, 128, 157,  78, 105, 134], dtype=int64)

In [15]:
df.replacement_cost.unique()

array([20.99, 12.99, 18.99, 26.99, 22.99, 17.99, 28.99, 15.99, 21.99,
       24.99, 16.99, 23.99, 10.99, 14.99, 27.99,  9.99, 19.99, 11.99,
       29.99, 25.99, 13.99])

In [16]:
df.rating.unique()

array(['PG', 'G', 'NC-17', 'PG-13', 'R'], dtype=object)

As it is explained in the following webpage: https://www.showbizjunkies.com/mpaa-ratings/


***G — General Audiences*** --> All Ages Admitted

***PG — Parental Guidance Suggested*** -->Some Material May Not Be Suitable For Children

***PG-13 — Parents Strongly Cautioned***  --> Some Material May Be Inappropriate For Children Under 13

***R — Restricted*** --> Children Under 17 Require Accompanying Parent or Adult Guardian

***NC-17 — No One 17 and Under Admitted*** 

In [17]:
df.special_features.unique()

array(['Deleted Scenes,Behind the Scenes', 'Trailers,Deleted Scenes',
       'Commentaries,Behind the Scenes', 'Deleted Scenes', 'Trailers',
       'Commentaries,Deleted Scenes',
       'Trailers,Deleted Scenes,Behind the Scenes',
       'Trailers,Commentaries,Behind the Scenes', 'Trailers,Commentaries',
       'Trailers,Behind the Scenes',
       'Commentaries,Deleted Scenes,Behind the Scenes',
       'Trailers,Commentaries,Deleted Scenes',
       'Trailers,Commentaries,Deleted Scenes,Behind the Scenes',
       'Behind the Scenes', 'Commentaries'], dtype=object)

In [18]:
df.last_update.unique()

array(['2006-02-15 05:03:42'], dtype=object)

In [19]:
df.last_update=pd.to_datetime(df.last_update)

In [20]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   film_id           1000 non-null   int64         
 1   title             1000 non-null   object        
 2   description       1000 non-null   object        
 3   release_year      1000 non-null   int64         
 4   language_id       1000 non-null   int64         
 5   rental_duration   1000 non-null   int64         
 6   rental_rate       1000 non-null   float64       
 7   length            1000 non-null   int64         
 8   replacement_cost  1000 non-null   float64       
 9   rating            1000 non-null   object        
 10  special_features  1000 non-null   object        
 11  last_update       1000 non-null   datetime64[ns]
dtypes: datetime64[ns](1), float64(2), int64(5), object(4)
memory usage: 93.9+ KB


In [21]:
df.head()

film_id             title  \
0        1  ACADEMY DINOSAUR   
1        2    ACE GOLDFINGER   
2        3  ADAPTATION HOLES   
3        4  AFFAIR PREJUDICE   
4        5       AFRICAN EGG   

                                         description  release_year  \
0  A Epic Drama of a Feminist And a Mad Scientist...          2006   
1  A Astounding Epistle of a Database Administrat...          2006   
2  A Astounding Reflection of a Lumberjack And a ...          2006   
3  A Fanciful Documentary of a Frisbee And a Lumb...          2006   
4  A Fast-Paced Documentary of a Pastry Chef And ...          2006   

   language_id  rental_duration  rental_rate  length  replacement_cost rating  \
0            1                6         0.99      86             20.99     PG   
1            1                3         4.99      48             12.99      G   
2            1                7         2.99      50             18.99  NC-17   
3            1                5         2.99     117             26.99      G   
4            1                6         2.99     130             22.99      G   

                   special_features         last_update  
0  Deleted Scenes,Behind the Scenes 2006-02-15 05:03:42  
1           Trailers,Deleted Scenes 2006-02-15 05:03:42  
2           Trailers,Deleted Scenes 2006-02-15 05:03:42  
3    Commentaries,Behind the Scenes 2006-02-15 05:03:42  
4                    Deleted Scenes 2006-02-15 05:03:42

In [22]:
df.drop_duplicates().shape==df.shape # no hay duplicados

True

In [23]:
df.drop("last_update", axis=1, inplace=True)

In [24]:
df.head()

film_id             title  \
0        1  ACADEMY DINOSAUR   
1        2    ACE GOLDFINGER   
2        3  ADAPTATION HOLES   
3        4  AFFAIR PREJUDICE   
4        5       AFRICAN EGG   

                                         description  release_year  \
0  A Epic Drama of a Feminist And a Mad Scientist...          2006   
1  A Astounding Epistle of a Database Administrat...          2006   
2  A Astounding Reflection of a Lumberjack And a ...          2006   
3  A Fanciful Documentary of a Frisbee And a Lumb...          2006   
4  A Fast-Paced Documentary of a Pastry Chef And ...          2006   

   language_id  rental_duration  rental_rate  length  replacement_cost rating  \
0            1                6         0.99      86             20.99     PG   
1            1                3         4.99      48             12.99      G   
2            1                7         2.99      50             18.99  NC-17   
3            1                5         2.99     117             26.99      G   
4            1                6         2.99     130             22.99      G   

                   special_features  
0  Deleted Scenes,Behind the Scenes  
1           Trailers,Deleted Scenes  
2           Trailers,Deleted Scenes  
3    Commentaries,Behind the Scenes  
4                    Deleted Scenes

In [25]:
df.to_csv(r'C:\Users\alexd\Ironhack\Proyectos\Project2_DA_Ironhack\Cleaned_data\film_cleaned.csv', index=False)